File: goApi/util/helper/response.go
package helper

import (
	"net/http"
)

const (
	StatusSuccessString = "success"
	StatusErrorString   = "error"
)

type Response struct {
	Code    int16 `json:"code"`
	ErrCode int64 `json:"errCode"`
	Status  string `json:"status"`
	Msg     string `json:"msg"`
	Data    interface{}   `json:"data"`
}

func RespSuccess(message string, data interface{}) Response {
	var resp Response
	resp.Code = http.StatusOK
	resp.ErrCode = 0
	resp.Status = StatusSuccessString
	resp.Msg = message
	resp.Data = data
	return resp
}

func RespError(message string, errCode int64, data interface{}) Response {
	var resp Response
	resp.Code = http.StatusOK
	resp.ErrCode = errCode
	resp.Status = StatusErrorString
	resp.Msg = message
	resp.Data = data
	return resp
}

File: phpFileSys/App/Common/Util/Response.php
<?php

namespace App\Common\Util;

class Response
{

   private int $code =200;

   private int $msg;

    /**
     * @var mixed
     */
   private  $data=[];

   private string $status='';

   private int $errCode=0;

    /**
     * @param string $msg
     * @param $data
     * @return string
     */
    public function success(string $msg,$data ): string{
        $this->code = http_response_code();

        $this->msg = $msg ?? 'success';

        $this->data = $data ?? [];

        $this->status = 'success';

        $this->errCode =0;

        return $this->toString();
    }


    /**
     * @param string $msg
     * @param int $errCode
     * @param $data
     * @return string
     */
    public function fail(string $msg, int $errCode, $data): string
    {
        $this->code = 400;

        $this->msg = $msg ?? 'error';

        $this->data = $data ?? [];

        $this->status = 'error';

        $this->errCode = $errCode ?? 500;

        return $this->toString();
    }

    public function toString(){
        return json_encode([
            'code' => $this->code,

            'msg' => $this->msg,

            'data' => $this->data,

            'status' => $this->status,

            'errCode' => $this->errCode,
        ]);
    }
}